backend/corpora/common/utils/regex.py

```diff
@@ -0,0 +1,7 @@
+USERNAME_REGEX = r"(?P<username>[\w\-\|]+)"
+UUID_REGEX = r"[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}"
+EXTENSION_REGEX = r"(?P<extension>h5ad)"
+DATASET_ID_REGEX = f"(?P<dataset_uuid>{UUID_REGEX})"
+COLLECTION_ID_REGEX = f"(?P<collection_uuid>{UUID_REGEX})"
+CURATOR_TAG_NAME_REGEX = r"(?P<tag>.*)"
```

Comment:
```suggestion
CURATOR_TAG_NAME_REGEX = r"(?P<tag>.*\.h5ad)"
```

Reply:
extension should be part of the curator tag regex. 

I'd prefer we not introduce this concept of the "curator tag _name_", i.e. the tag with the extension removed, unless it is absolutely necessary. Because it just adds to confusion around what constitutes the curator tag entity.

Reply:
I'll change the name.